backend/prisma/schema.prisma

```diff
@@ -11,64 +8,76 @@ datasource db {
 }
 
 model Poll {
-  id                String      @id @default(uuid()) @db.Uuid
-  createdAt         DateTime    @default(now())
-  updatedAt         DateTime    @updatedAt
-  name              String
-  creatorId         String      @db.Uuid
-  creator           User?       @relation("PollUser", fields: [creatorId], references: [id])
-  adminLink         String
-  pollLink          String
-  resultLink        String
-  questions         Question[]  @relation(name: "PollQuestions")
-  isActive          Boolean     @default(true)
+  id         String     @id @default(uuid()) @db.Uuid
+  createdAt  DateTime   @default(now())
+  updatedAt  DateTime   @updatedAt
+  name       String
+  creatorId  String     @db.Uuid
+  adminLink  String
+  pollLink   String
+  resultLink String
+  isActive   Boolean    @default(true)
+  creator    User       @relation("PollUser", fields: [creatorId], references: [id])
+  questions  Question[] @relation("PollQuestions")
 }
 
 model Question {
-  id                String      @id @default(uuid()) @db.Uuid
-  createdAt         DateTime    @default(now())
-  updatedAt         DateTime    @updatedAt
-  pollId            String      @db.Uuid
-  poll              Poll        @relation(name: "PollQuestions", fields: [pollId], references: [id])
-  typeId            String      @db.Uuid
-  type              Type        @relation(name: "QuestionType", fields: [typeId], references: [id])
-  options           Option[]    @relation(name: "QuestionOptions")
-  votes             Vote[]      @relation(name: "QuestionVotes")
+  id           String     @id @default(uuid()) @db.Uuid
+  createdAt    DateTime   @default(now())
+  updatedAt    DateTime   @updatedAt
+  pollId       String     @db.Uuid
+  typeId       String     @db.Uuid
+  maxValue     Float?
+  minValue     Float?
+  parentId     String?    @db.Uuid
+  step         Float?
+  typeName     String     @default("free")
+  description  String     @default("")
+  title        String     @default("")
+  parent       Question?  @relation("SubQuestions", fields: [parentId], references: [id])
+  poll         Poll       @relation("PollQuestions", fields: [pollId], references: [id])
+  type         Type       @relation("QuestionType", fields: [typeId], references: [id])
+  options      Option[]   @relation("QuestionOptions")
+  subQuestions Question[] @relation("SubQuestions")
+  votes        Vote[]     @relation("QuestionVotes")
 }
 
 model Type {
```

Comment:
Do we need to have a Type table anymore as we have the typeName? Could this be removed?

Reply:
Sure, the Type table could be removed. I thought about removing it but left it there in case you had some kind of plan for it.

Reply:
Well, I don't have any use for it so it can be removed as part of this PR or we can create an another issue for removing it

Reply:
Probably not a good idea to remove it now, since it would break some code. Since type is a required field, an existing dummy type is currently used.

Reply:
Ok, I will create a new issue for this.